### ISSUE TITLE
Error running agent: retrieving resolver failed: linking secrets failed when secretstore dynamic config is set to true

### ISSUE THREAD

**Issue by kishorsn77** [influxdata/telegraf#13345]
### Relevant telegraf.conf

```toml
# # Operating System native secret-store
 [[secretstores.os]]
#   ## Unique identifier for the secret-store.
#   ## This id can later be used in plugins to reference the secrets
#   ## in this secret-store via @{<id>:<secret_key>} (mandatory)
   id = "secretstore"
# 
#   ## Keyring name used for the secrets
#   # keyring = "telegraf"
# 
#   ## Allow dynamic secrets that are updated during runtime of telegraf
   dynamic = true
```

### Logs from Telegraf

```text
2023-05-26T12:28:57Z E! [telegraf] Error running agent: retrieving resolver failed: linking secrets failed: unlinked part "@{secretstore:snmpTrap_SecurityName}"
```

### System info

Telegraf 1.25.1, Operating system: Windows & Linux

### Steps to reproduce

1. Set the secretstore plugin in the conf file with dynamic = true
2. Use the secret in any of the input plugins
3. Run the telegraf service

...

### Expected behavior

Telegraf should start normally

### Actual behavior

- Telegraf does not start
- No Log file is generated
- Error shown only when telegraf is run manually from command line in both Windows & Linux

### Additional info

Telegraf works normally when dynamic = false

**Comment by powersj** [comment]
Hi,

> Run the telegraf service

This is most likely your problem. The user you set up the secrets as and the user who runs the service (e.g. `telegraf`) are probably different and not able to see the secret you created.

There should be more to that error message that you may not have included. For example, if the secret was not set up it should say:

```s
2023-05-26T18:50:17Z E! retrieving resolver failed: linking secrets failed: resolving "@{secretstore:snmpTrap_SecurityName}" failed: The specified item could not be found in the keyring
```

**Comment by kishorsn77** [comment]
@powersj - I am setting and retrieving the secrets with the same user. Everything works fine if I set **dynamic = false** in the telegraf.conf file under secrets.os plugin. 
Telegraf won't even start if I set "dynamic = true". Unfortunately, though I have set telegraf in debug mode, I see only the error message which I shared.
One more update. I tried running the same with 1.26.3 version(latest). Everything works fine as expected. Looks like the issue is with 1.25.1 version.
